addons/crm/tests/test_crm_lead.py

```diff
@@ -158,6 +159,59 @@ def test_crm_lead_partner_sync(self):
         self.assertFalse(partner.email_normalized)
         self.assertFalse(partner.phone)
 
+    @users('user_sales_manager')
+    def test_crm_lead_partner_sync_email_phone(self):
+        """ Specifically test synchronize between a lead and its partner about
+        phone and email fields. Phone especially has some corner cases due to
+        automatic formatting (notably with onchange in form view). """
+        lead, partner = self.lead_1.with_user(self.env.user), self.contact_2
+        lead_form = Form(lead)
+
+        # reset partner phone to a local number
+        partner_phone, partner_email = '202 555 0999', partner.email
+        partner_phone_formatted = phone_format(partner_phone, 'US', '1')
+        partner_phone_sanitized = phone_format(partner_phone, 'US', '1', force_format='E164')
+        self.assertEqual(partner_phone_formatted, '+1 202-555-0999')
+        self.assertEqual(partner_phone_sanitized, '+12025550999')
+        partner.phone = partner_phone
+
+        # email & phone must be automatically set on the lead
+        lead_form.partner_id = partner
+        self.assertEqual(lead_form.email_from, partner_email)
+        self.assertEqual(lead_form.phone, partner_phone_formatted)
+        self.assertFalse(lead_form.ribbon_message)
+
+        lead_form.save()
+        self.assertEqual(partner.phone, partner_phone)
+        self.assertEqual(lead.phone, partner_phone_formatted)
+        self.assertEqual(lead.phone_sanitized, partner_phone_sanitized)
+
+        # writing on the lead field must change the partner field
+        new_phone = '+1 202 555 7799'
+        new_phone_formatted = phone_format(new_phone, 'US', '1')
+        lead_form.email_from = '"John Zoidberg" <john.zoidberg@test.example.com>'
+        self.assertIn('the customer email', lead_form.ribbon_message)
+        lead_form.phone = new_phone
+        self.assertEqual(lead_form.phone, new_phone_formatted)
+        self.assertIn('the customer email', lead_form.ribbon_message)
```

Comment:
Should be testing "the customer phone" here, right?

Reply:
Yes sir.